onnx/defs/rnn/defs.cc

```diff
@@ -6,125 +6,206 @@
+	   "`[num_directions, 3*hidden_size, input_size]`.", "T")
+    .Input(2, "R",
+	   "The recurrence weight tensor. Concatenation of `R[zrh]` and `RB[zrh]` "
+	   "(if bidirectional) along dimension 0. This tensor has shape "
+	   "`[num_directions, 3*hidden_size, hidden_size]`.", "T")
+    .Input(3, "B",
+	   "The bias tensor for the gates. Concatenation of `[Wb[zrh], Rb[zrh]]` and "
+           "`[WBb[zrh], RBb[zrh]]` (if bidirectional) along dimension 0. This tensor "
+           "has shape `[num_directions, 6*hidden_size]`. Optional: If not specified "
+           "- assumed to be 0", "T",
+	   true /*optional*/)
+    .FillUsing(RNNDocGenerator("GRU"));
 
-Note, that for LSTM and 2 out of 3 gates for GRU, there are duplicate biases for
-the gates (model is overparametrized). It follows CuDNN/TensorRT convention and
-allows to make spec more uniform.
-)DOC", AttrType::STRING, true)
-    .Attr("directions",
-          "Number of directions: 1 for unidirectional (default) and 2 for "
-          "bidirectional",
-          AttrType::INT)
-    .Attr("skip_input_transform",
-          "If set, skips linear transformation on the input of the first layer",
-          AttrType::INT)
-    .Attr("num_layers", "Numbers of RNN layers in the stack, default 1",
-          AttrType::INT)
-    .Attr("hidden_size", "Number of neurons in the hidden layer", AttrType::INT)
-    .Input(0, "weights", R"DOC(
-All parameters of the stack packed together in the opaque tensor. The size must
-be compatible with input attributes passed to the op.
 
-The layout format is the one used by CuDNN and very similar to TensorRT:
+OPERATOR_SCHEMA(LSTM)
+    .NumInputs(3, 8)
+    .NumOutputs(1, 2)
+    .SetDoc(R"DOC(
+Computes an one-layer LSTM. This operator is usually supported via some
+custom implementation such as CuDNN.
 
-The weight structure holds weights and biases for each layer of the network.
-Each parameter matrix is linearly appended after the previous parameter matrix
-without padding.
+Notations:
+`X` - input tensor
+`i` - input gate
+`o` - output gate
+`f` - forget gate
+`c` - cell gate
+`t` - time step (t-1 means previous time step)
+`W[iofc]` - W parameter weight matrix for input, output, forget, and cell gates
+`R[iofc]` - R recurrence weight matrix for input, output, forget, and cell gates
+`Wb[iofc]` - W bias vectors for input, output, forget, and cell gates
+`Rb[iofc]` - R bias vectors for input, output, forget, and cell gates
+`P[iof]`  - P peephole weight matrix for input, output, and forget gates
+`WB[iofc]` - W parameter weight matrix for backward input, output, forget, and cell gates
+`RB[iofc]` - R recurrence weight matrix for backward input, output, forget, and cell gates
+`WBb[iofc]` - W bias vectors for backward input, output, forget, and cell gates
+`RBb[iofc]` - R bias vectors for backward input, output, forget, and cell gates
+`PB[iof]`  - P peephole weight vector for backward input, output, and forget gates
```

Comment:
denote that it's optional

Reply:
This is for notation. We make it clear it is optional when talking about input `P`.